gcloud/datastore/client.py

```diff
@@ -288,9 +289,10 @@ def get_multi(self, keys, missing=None, deferred=None):
         if not keys:
             return []
 
-        ids = list(set([key.dataset_id for key in keys]))
-        if ids != [self.dataset_id]:
-            raise ValueError('Keys do not match dataset ID')
+        ids = set(key.dataset_id for key in keys)
+        for ds_id in ids:
```

Comment:
Agreed, I'll change that. `current_id` sounds perfect to make it clear we're talking about the iterator and not the attribute